core/app/models/spree/product.rb

```diff
@@ -105,7 +105,7 @@ class Product < Spree::Base
       validates :price, if: proc { Spree::Config[:require_master_price] }
     end
 
-    validates :slug, length: { minimum: 3 }, allow_blank: true, uniqueness: true
+    validates :slug, allow_blank: true, uniqueness: true
```

Comment:
```ruby
validates :slug, uniqueness: { allow_blank: true }, presence: true
```

Here's the output from the console (p is persisted product):

>[14] pry(main)> p.slug = ""
=> ""
[15] pry(main)> p.valid?
=> false
[16] pry(main)> p.slug = nil
=> nil
[17] pry(main)> p.valid?
=> true
[18] pry(main)> p.slug
=> "asd-product-c7a2f169-89c1-48a2-a378-2de2cce099af"

So when slug is an empty string, validation fails. When slug is nil, friendly_id generates a new slug. (and since I already have a product with "asd-product" slug in my database, don't ask, it's very important product, it appends UUID (I think it's UUID))